packages/flutter_tools/lib/src/migrations/xcode_thin_binary_build_phase_input_paths_migration.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/project_migrator.dart';
+import '../xcode_project.dart';
+
+// Migrate Xcode Thin Binary build phase to use Info.plist from build directory
+// as an input file to avoid mDNS error.
```

Comment:
```suggestion
// Migrate Xcode Thin Binary build phase to depend on Info.plist from build directory
// as an input file to ensure it has been created before inserting the NSBonjourServices key
// to avoid an mDNS error.
```